pkg/cmd/audit.go

```diff
@@ -1 +1,133 @@
 package cmd
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericclioptions/resource"
+
+	"github.com/openshift/must-gather/pkg/util"
+)
+
+var (
+	auditExample = `
+	# Print all audit log events formatted:
+	%[1]s events https://<ci-artifacts>/masters-kube-apiserver-audit.log.gz
+
+	# Search for all audit events that include 'openshift-kube-apiserver-operator' in URI
+	%[1]s events https://<ci-artifacts>/masters-kube-apiserver-audit.log.gz --contain=openshift-kube-apiserver-operator
+
+	# Search for GET audit events that matches regexp 'apiserver-operator$' in URI
+	%[1]s events https://<ci-artifacts>/masters-kube-apiserver-audit.log.gz --regexp="apiserver-operator$" --verb=get
+`
+)
+
+type AuditOptions struct {
+	fileWriter *util.MultiSourceFileWriter
+	builder    *resource.Builder
+	args       []string
+
+	auditFileURL string
+	contains     string
+	regexp       string
+	verb         string
+
+	genericclioptions.IOStreams
+}
+
+func NewAuditOptions(streams genericclioptions.IOStreams) *AuditOptions {
+	return &AuditOptions{
+		IOStreams: streams,
+	}
+}
+
+func NewCmdAudit(parentName string, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewAuditOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:          "audit <URL> [flags]",
+		Short:        "Inspects the audit logs captured during CI test run.",
+		Example:      fmt.Sprintf(inspectExample, parentName),
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.Complete(c, args); err != nil {
+				return err
+			}
+			if err := o.Validate(); err != nil {
+				return err
+			}
+			if err := o.Run(); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVar(&o.contains, "contains", "", "Search for audit logs that contains specified URI")
+	cmd.Flags().StringVar(&o.regexp, "regexp", "", "Search for audit logs with URI that matches the regexp")
+	cmd.Flags().StringVar(&o.verb, "verb", "", "Filter result of search to only contain the specified verb (eg. 'update', 'get', etc..)")
+
+	return cmd
+}
+
+func (o *AuditOptions) Complete(command *cobra.Command, args []string) error {
+	if len(args) == 1 {
```

Comment:
You should error out when the URL wasn't passed, it's pointless when it's not there.